### PR TITLE
scoring bugfixes

### PR DESCRIPTION
**Pull request by achen2401**
A number of issues found with scoring display.

Fixes include:

- should display/plot score of **0** (see story [here](https://www.pivotaltracker.com/story/show/184197546))
- fix incorrect alert display for GDS score - correctly assigned a score of 0 when user answers for questions where a `no` should receive a score of 0 (questions, 2, 3, 4, 6, 8, 9, 10, 12, 14, 15) (see story [here](https://www.pivotaltracker.com/story/show/184155696))
- C-IDAS - display alert when user endorsed suicide ideation (see story [here](https://www.pivotaltracker.com/story/show/184155350))
- ECOG12 and CP-ECOG  an answer of I don't know (with numeric value of 9) shouldn't be added to the total score (see story [here](https://www.pivotaltracker.com/story/show/184200191))
- red exclamation icon will appear if the score severity is determined to be `high`: 
  BEHAV5 - score > 0
  C-IDAS - score > 18 and suicide question score >= 1
  GAD7 - score > 15
  GDS - score > 4
  MINICOG - score 0, 1, 2, 3
  SLUMS - score <= 19 (low education), score <= 20 (higher education)

Other changes:
- move alert scoring check from JS code to CQL for cleaner code and ease of management

### CODE REVIEW

File: src/components/Score.js [hunk 1]
```diff
@@ -1,38 +1,49 @@
...
-      ? qConfig[instrumentId].failedScores(scoreParams)
-      : [];
-  };
-  const arrFailedScores = getFailedScoringByInstrumentId(instrumentId);
+  const { score, justifyContent, alignItems, scoreParams } = props;
+  const scoreSeverity =
+    scoreParams && scoreParams.scoreSeverity
+      ? String(scoreParams.scoreSeverity).toLowerCase()
+      : null;
+  const arrAlertSeverityLevels = ["high", "moderately high", "moderate"];
```
Comment (ivan-c) [inline src/components/Score.js:14]: Does `low` need to be in this array? I saw some references in the CQL to `low`, but wasn't sure if it was the same

**Review by ivan-c — APPROVED** [review]
Looks great!!

File: src/components/Score.js (hunk 1 above)
Comment (achen2401) [reply src/components/Score.js]: thanks @ivan-c for catching this.  I have fixed so the code will only display alert icon for score that has severity level of `high` for now.  I renamed the variable and comment for clarity.  For now, the code will not display alert for severity of `moderate` / `moderately high` until specified by the project team.

File: src/cql/source/C-IDAS_InterventionLogicLibrary.cql [hunk 2]
```diff
@@ -55,33 +56,41 @@ define QuestionnaireResponses:
...
+            Coalesce(First(LogicHelpers.getScoringByResponseItem(CurrentQuestionnaire, I.item, question17LinkId)), 0) +
+            Coalesce(First(LogicHelpers.getScoringByResponseItem(CurrentQuestionnaire, I.item, question18LinkId)), 0)
+          else null,
+      suicideScore: Coalesce(First(LogicHelpers.getScoringByResponseItem(CurrentQuestionnaire, I.item, suicideQuestionLinkId)), 0)
+      return {
+        id: I.id.value,
+        date: LogicHelpers.DateTimeText(I.authored),
+        responses: LogicHelpers.FormattedResponses(QuestionnaireItems, I.item, ScoringQuestionId),
+        score: score,
+        scoreSeverity: if score > 18 or (suicideScore >= 1) then 'high' else 'low',
```
Comment (mcjustin) [inline src/cql/source/C-IDAS_InterventionLogicLibrary.cql:88]: @sierramatice  can you please review each of these instrument CQL files to see that the score categories align with spec?

**Review by sierramatice — APPROVED** [review]
@achen2401 I read through the code and it all looks good! I do want to note that the DCW team might be changing these cut off points in the future, but it currently matches the spec

**Review by mcjustin — APPROVED** [review]

**Pull request merged**